core/src/apps/management/recovery_device/recover.py

```diff
@@ -44,7 +44,15 @@ def process_slip39(words: str) -> Tuple[Optional[bytes], int, int]:
         )
         storage.recovery_shares.set(index_with_group_offset, words)
 
-        return None, share.group_index, share.index  # we need more shares
+        # if share threshold and group threshold are 1
+        # we can calculate the secret right away
+        if share.threshold == 1 and share.group_threshold == 1:
+            identifier, iteration_exponent, secret, _ = slip39.combine_mnemonics(
+                storage.recovery_shares.fetch()
```

Comment:
Any reason to fetch them from storage if we have them in the `words` variable?

Reply:
fixed.